src/syskit.ts

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+import * as autoproj from './autoproj';
+import * as context from './context';
+import * as wrappers from './wrappers';
+import * as rest from 'node-rest-client';
+import { CancellationToken } from 'vscode'
+
+export class DeployedTask
+{
+    task_name : string;
+    task_model_name : string;
+}
+
+export class AvailableDeployment
+{
+    name: string;
+    project_name: string;
+    tasks: DeployedTask[] = [];
+    default_deployment_for: string | undefined;
+    default_logger: string | undefined;
+}
+
+export class RegisteredDeploymentInfo
+{
+    id : number;
+    created : boolean;
+    deployment_name: string;
+    tasks : DeployedTask[];
+    on : string;
+    mappings : Map<string, string>;
+    type : string;
+}
+
+export class CommandLine
+{
+    env: object;
+    command : string;
+    args: string[];
+    working_directory: string;
+};
+
+
+export class Connection
+{
+    private _workspace : autoproj.Workspace;
+    private _client : rest.Client;
+
+    private _host : string;
+    private _port : number;
+
+    constructor(workspace : autoproj.Workspace,
+        host : string = 'localhost',
+        port : number = 20202)
+    {
+        this._workspace = workspace;
+        this._client    = new rest.Client();
+        this._host      = host;
+        this._port      = port;
+    }
+
+    private callBase(method : string, expectedStatus : number, path : string) : Promise<any>
+    {
+        return new Promise((resolve, reject) => {
+            let url = `http://${this._host}:${this._port}/api/syskit/${path}`;
+            this._client[method](url, function (data, response) {
+                if (response.statusCode != expectedStatus) {
+                    let msg = data.error || data;
+                    reject(new Error(`${method} ${url} error: ${msg}`));
```

Comment:
VSCode's ui for showing error messages is not very useful with long messages. I would've shown only `${msg}`

Reply:
> VSCode's ui for showing error messages is not very useful with long messages. I would've shown only ${msg}

I did it this way since these should be internal errors (the UI should avoid error cases), and it will *really* help us developers if the user can copy/paste the exact URL.

Reply:
I can actually see the whole error message if I hover it for a while (which I didn't realize until after I made the comment) so I guess this is fine 👍